client/components/chart/x-axis.jsx

```diff
@@ -4,91 +4,69 @@
  * External dependencies
  */
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useLayoutEffect, useState, useRef } from 'react';
 import { numberFormat } from 'i18n-calypso';
 
 /**
  * Internal dependencies
  */
-import afterLayoutFlush from 'lib/after-layout-flush';
 import Label from './label';
 
-export default class ChartXAxis extends React.PureComponent {
-	static displayName = 'ModuleChartXAxis';
-
-	static propTypes = {
-		data: PropTypes.array.isRequired,
-		isRtl: PropTypes.bool,
-		labelWidth: PropTypes.number.isRequired,
-	};
-
-	axisRef = React.createRef();
-	axisSpacerRef = React.createRef();
-
-	state = {
-		divisor: 1,
-		spacing: this.props.labelWidth,
-	};
-
-	// Add listener for window resize
-	componentDidMount() {
-		this.resize = afterLayoutFlush( this.resize );
-		window.addEventListener( 'resize', this.resize );
-		this.resize();
-	}
-
-	// Remove listener
-	componentWillUnmount() {
-		if ( this.resize.cancel ) {
-			this.resize.cancel();
+const ModuleChartXAxis = ( { data, isRtl, labelWidth } ) => {
+	const axisRef = useRef( null );
+	const axisSpacerRef = useRef( null );
+	const dataCount = data.length || 1;
+
+	const [ spacing, setSpacing ] = useState( labelWidth );
+
+	const [ divisor, setDivisor ] = useState( 1 );
+
+	useLayoutEffect( () => {
+		const resize = () => {
+			const width = axisRef.current.clientWidth - axisSpacerRef.current.clientWidth;
```

Comment:
Non-blocking, but should we throttle the resize event? Accessing clientWidth triggers a browser reflow which is a bit expensive.